src/robot/libraries/Collections.py

```diff
@@ -387,7 +387,12 @@ def lists_should_be_equal(self, list1, list2, msg=None, values=True,
         default = 'Lengths are different: %d != %d' % (len1, len2)
         _verify_condition(len1 == len2, default, msg, values)
         names = self._get_list_index_name_mapping(names, len1)
-        diffs = list(self._yield_list_diffs(list1, list2, names))
+        if not ignore_order:
+            diffs = list(self._yield_list_diffs(list1, list2, names))
+        else:
+            sorted_list1 = sorted(list1)
+            sorted_list2 = sorted(list2)
+            diffs = list(self._yield_list_diffs(sorted_list1, sorted_list2, names))
```

Comment:
This looks a bit more complicated than needed. AFAIK this would be enough:

```python
if ignore_order:
    list1 = sorted(list1)
    list2 = sorted(list2)
```